modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -306,13 +324,35 @@ public CommerceOrder addCommerceOrder(
 		commerceOrder.setCompanyId(user.getCompanyId());
 		commerceOrder.setUserId(userId);
 		commerceOrder.setUserName(user.getFullName());
-		commerceOrder.setBillingAddressId(billingAddressId);
 		commerceOrder.setCommerceAccountId(commerceAccountId);
 		commerceOrder.setCommerceCurrencyCode(commerceCurrencyCode);
 		commerceOrder.setCommerceOrderTypeId(commerceOrderTypeId);
 		commerceOrder.setCommerceShippingMethodId(commerceShippingMethodId);
-		commerceOrder.setShippingAddressId(shippingAddressId);
-		commerceOrder.setCommercePaymentMethodKey(commercePaymentMethodKey);
+
+		if (billingAddressId > 0) {
+			commerceOrder.setBillingAddressId(billingAddressId);
+		}
+		else {
+			_setAccountDefaultBillingAddress(commerceChannel, commerceOrder);
+		}
+
+		if (shippingAddressId > 0) {
+			commerceOrder.setShippingAddressId(shippingAddressId);
+		}
+		else {
+			_setAccountDefaultShippingAddress(commerceChannel, commerceOrder);
+		}
+
+		if (Validator.isNotNull(commercePaymentMethodKey)) {
+			commerceOrder.setCommercePaymentMethodKey(commercePaymentMethodKey);
+		}
+		else {
+			_setAccountDefaultPaymentIntegrationKey(
+				commerceChannel, commerceOrder);
+		}
```

Comment:
setting of `billingAddressId` and `shippingAddressId` needs to ocme after `setCommerceAccountId`.

setting of `commercePaymentMethodKey` in `setAccountDefaultPaymentIntegrationKey` needs to come before setting of `paymentCommerceTermEntryId` in `_setAccountDefaultPaymentTerm`

The formatting of this was intentional so we keep the code clean look.